sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -1032,9 +1032,19 @@ object Hex {
     array
   }
 
+  private def maxArraySize(length: Long): Int = {
+    if (length > Int.MaxValue) {
+      throw QueryExecutionErrors.tooManyArrayElementsError(length, Int.MaxValue)
```

Comment:
What is the actual error message here, @yaooqinn ?

Reply:
Hi @dongjoon-hyun 
https://spark.apache.org/docs/4.0.0-preview1/sql-error-conditions.html#collection-size-limit-exceeded

Reply:
I saw that. However, I'm not sure this is `cannot initialize an array with specified parameters.` or not. When you check it manually, what was the error message, @yaooqinn ?

Reply:
I've lost my local history. Let me rebuild and rerun in order to identify and attach the error.

Reply:
Hi @dongjoon-hyun 
```scala
 org.apache.spark.sql.catalyst.expressions.Hex.hex((" " * (Int.MaxValue / 2 + 1)).getBytes)
org.apache.spark.SparkIllegalArgumentException: [COLLECTION_SIZE_LIMIT_EXCEEDED.INITIALIZE] Can't create array with 2147483648 elements which exceeding the array size limit 2147483647, cannot initialize an array with specified parameters. SQLSTATE: 54000
  at org.apache.spark.sql.errors.QueryExecutionErrors$.tooManyArrayElementsError(QueryExecutionErrors.scala:2517)
  at org.apache.spark.sql.catalyst.expressions.Hex$.hex(mathExpressions.scala:1042)
  ... 42 elided
```